QRScanner.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react'
+import {
+    Text, 
+    View, 
+    StyleSheet,
+    Dimensions,
+    Alert
+} from 'react-native'
+import { BarCodeScanner, Permissions } from 'expo'
+import CodeModel from './Models/Code'
+
+export default class QRScanner extends React.Component {
+
+    /**
+     * Set default state with the camera permission status and whether or not an alert is showing.
+     */
+    state = {
```

Comment:
In web react, we do this in the `constructor` function. Does that exist in react native?